### PR TITLE
fix(update) ensure alpine and rhel update correctly

### PR DESCRIPTION
**Pull request by gszr**
Also update kbt version in makefile to match kong's -- and add automation to `update.sh`

### CODE REVIEW

File: alpine/Dockerfile [hunk 1]
```diff
@@ -12,11 +12,17 @@ COPY kong.tar.gz /tmp/kong.tar.gz
 ARG KONG_VERSION=2.2.1
 ENV KONG_VERSION $KONG_VERSION
 
+ARG KONG_AMD64_SHA="e05c7075ae263bb7ef84b0370bf4e31f5246a41d910e5d4336216ae95b654af1"
+ENV KONG_AMD64_SHA $KONG_AMD64_SHA
+
+ARG KONG_ARM64_SHA="c7b2c0e1f47e3009e55b492344bd1ff689a41eca91c21f2c3f57f9172932c96d"
+ENV KONG_ARM64_SHA $KONG_ARM64_SHA
```
Comment (hutchic) [inline alpine/Dockerfile:19]: I think this is unsupported in the official docker build system

File: update.sh [hunk 2]
```diff
@@ -23,12 +23,22 @@ function die() {
 
 hub --version &> /dev/null || die "hub is not in PATH. Get it from https://github.com/github/hub"
 
+kbt_in_kong_v=$(curl -sL https://raw.githubusercontent.com/Kong/kong/$version/Makefile | grep 'KONG_BUILD_TOOLS\s*?=' | awk -F"=" '{print $2}' | tr -d "'[:space:]")
+sed -i -e 's/KONG_BUILD_TOOLS?=.*/KONG_BUILD_TOOLS?='$kbt_in_kong_v'/g' Makefile
```
Comment (hutchic) [inline update.sh:27]: clever 👍

File: update.sh [hunk 3]
```diff
@@ -23,12 +23,22 @@ function die() {
...
-   sed -i -e 's/ARG KONG_SHA256=.*/ARG KONG_SHA256=\"'$new_sha'\"/g' Dockerfile
+
+   sed -i -e 's/ARG KONG_AMD64_SHA=.*/ARG KONG_AMD64_SHA=\"'$new_sha'\"/g' Dockerfile
+   sed -i -e 's/ARG KONG_VERSION=.*/ARG KONG_VERSION='$version'/g' Dockerfile
+
+   url=$(grep bintray.com Dockerfile | awk -F" " '{print $3}' | sed 's/\"//g' | sed 's/$KONG_VERSION/'$version'/g' | sed 's/$arch/'arm64'/g')
+   curl -fL $url -o /tmp/kong
+   new_sha=$(sha256sum /tmp/kong | cut -b1-64)
+
+   sed -i -e 's/ARG KONG_ARM64_SHA=.*/ARG KONG_ARM64_SHA=\"'$new_sha'\"/g' Dockerfile
```
Comment (hutchic) [inline update.sh:41]: full disclosure: this looks correct but I didn't actually test it 👍

**Review by hutchic — CHANGES_REQUESTED** [review]
we've had challenges in the past with using `ARG thing=default` with the official docker repository

File: alpine/Dockerfile (hunk 1 above)
Comment (gszr) [reply alpine/Dockerfile:19]: @tianon Apologies to bug, but could you comment -- is this something unsupported by the Docker Library build system?

Comment (hutchic) [reply alpine/Dockerfile:19]: could be I'm thinking of
> Unfortunately our build systems cannot handle variables in COPY or FROM statements.

trying to locate a past PR where we had thing setup in a similar fashion to see why we didn't go forward with it

Comment (tianon) [reply alpine/Dockerfile:19]: IMO it's a bit "extra", but it's fine (and is done for `KONG_VERSION` just above this).

Comment (gszr) [reply alpine/Dockerfile:19]: Thanks for commenting, @tianon!

**Review by hutchic — APPROVED** [review]

**Pull request merged**